gensim/models/word2vec_inner.pyx

```diff
@@ -633,6 +632,9 @@ def train_sentence_sg(model, sentence, alpha, _work):
             else:
                 codelens[i] = 1
             result += 1
+    # single randint() call avoids a big thread-sync slowdown
+    for i, item in enumerate(np.random.randint(0, window, sentence_len)):
+        reduced_windows[i] = item
```

Comment:
If it's too hairy or obscure then never mind. I haven't actually tried / profiled that change, it's just a minor nitpick.